### ISSUE TITLE
Allow appending to existing build

### ISSUE THREAD

**Issue by andrzej-kodify** [percy/percy-storybook#51]
Since I have already set up Jenkins pipeline with percy.io for visual regression testing of the site I'd like to append storybook results to same build. 
It would be nice to have parameter or env variable where I could pass existing percy build id. Otherwise I end up having two separate percy builds per single jenkins build

I see the flow as follows:

1. run percy storybook
2a. if there's an existing build id passed don't create new build
2b. if there's no build id passed create new build (current behaviour)
3. add storybook results to build
4a. if there was build id passed just finish execution and don't finalize build
4b. if there wasn't build id passed finalize build (current behaviour)

**Comment by timhaines** [comment]
@andrzej-kodify Interesting suggestion.  What's your use case here?

If you have 2 different Storybooks, I'd recommend creating two projects in Percy (inside the your existing organization).  Our GitHub integration supports having multiple Percy projects report into the one GitHub repository/pull request.  You just need to make sure that the correct PERCY_TOKEN is provided for each storybook process when it's run in Jenkins.

**Comment by timhaines** [comment]
@andrzej-kodify Okay.  I'm a little confused.  Are you expecting the snapshots in the Percy build to be identical because they both come from the same Storybook build?  In that case would it make sense to disable one of the builds (perhaps with PERCY_ENABLE=0) so that it doesn't attempt to create the same snapshots twice?

Or are you expecting two different SDKs to be able to send snapshots to the one Percy project?

**Comment by andrzej-kodify** [comment]
Two different SDKs sounds a bit more like my case. But let me elaborate.
I've got single repository with my react website project and test automation using testcafe and percy-js.
In jenkins pipeline I:
1. build the website and storybook static files
2. deploy website to test environment
3. use precy-js to create new build
4. here I want to use percy-storybook to take and upload screenshots from storybook
5. launch testcafe. This basically opens and controls the browser doing functional tests. It also takes screenshots whenever I want them. Each screenshot is uploaded to percy via percy-js with a distinctive name (so it is nicely matched between builds and "same" screenshots are compared).
6. after all tests are complete I finalize the build using percy-js

So right now at point 4 I can either duplicate all the logic to create screenshots from storybook myself (which is reinventing the wheel since you guys are doing this) or I can launch precy-storybook anywhere in the pipeline but it will always be separate build in percy since storybook plugin always creates and finalizes build when executed.

Let me know if this still isn't clear :)

**Comment by timhaines** [comment]
Hi @andrzej-kodify - thank you, I understand your scenario now.  I think you'd have an easy time if you created a second Percy project.  Have percy-storybook run with the PERCY_TOKEN from project A, and have the testcafe snapshots run with the PERCY_TOKEN from project B.  Both of these projects can be linked to the same GitHub repo, and will both show up as separate statuses on the pull request.  I'm confident this solution would work well for you.  Could you give it a try and let us know?

**Comment by timhaines** [comment]
@andrzej-kodify I'm closing this for now as I haven't heard from you.  Feel welcome to contact support to discuss this further.

**Comment by GeorgeTaveras1231** [comment]
@timhaines I know this is old but I'm encountering a need for this. Similarly, we have multiple storybooks. In fact we have a monorepo with 12 storybook instances. Creating a percy project for each storybook instance will produce a lot of management overhead for us. And a lot of friction when stories get re-structured (moved from one storybook to another, instances are consolidated or broken up into separate storybooks). At the same time, we have screenshots that are produced by cypress (using the `@percy/cypress` package.

I think the position to not support this is inconsistent with other percy libraries. `@percy/cypress` for example, uses the [`postSnapshot`](https://www.npmjs.com/package/@percy/sdk-utils#postsnapshot) API from `@percy/sdk-utils` to send the snapshot information to the local running percy API server, and therefore allowing you to do run as many screenshot-producing processes as you want:

```bash
# Somewhere at the beginning of pipeline
percy exec start

// Run any number of processes that post snapshots to the local percy server (such as running multiple cypress processes in parallel)

# After all processes that generate snapshots end
percy exec stop
```

I think it would be great if we could use the percy-storybook CLI to add screenshots to existing projects. I'm maybe over simplifying it but it seems like this project just needs to conditionally run `percy.start` (based on some flag maybe), and use `postSnapshot` instead of `percy.snapshot`.
